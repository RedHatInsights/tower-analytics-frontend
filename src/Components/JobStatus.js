import React from 'react';
import PropTypes from 'prop-types';
import { Label } from '@patternfly/react-core';
import {
    CheckCircleIcon,
    ExclamationCircleIcon,
    ClockIcon,
    SyncAltIcon,
    TimesCircleIcon
} from '@patternfly/react-icons';

const JobStatus = ({ status }) => {
    const capitalized = (str) => str.charAt(0).toUpperCase() + str.slice(1);

    const getColor = () => {
        switch (status) {
            case 'successful':
                return 'green';
            case 'failed':
            case 'error':
                return 'red';
            case 'running':
                return 'blue';
            // case new, waiting, pending, canceled
            default: return 'grey';
        }
    };

    const getIcon = () => {
        switch (status) {
            case 'successful':
                return <CheckCircleIcon />;
            case 'failed':
            case 'error':
                return <ExclamationCircleIcon />;
            case 'running':
                return <SyncAltIcon />;
            case 'canceled':
                return <TimesCircleIcon />;
            // case new, waiting, pending
            default: return <ClockIcon />;
        }
    };

    return (
        <Label variant="outline" color={ getColor() } icon={ getIcon() }>{ capitalized(status) }</Label>
    );
};

JobStatus.propTypes = {
    status: PropTypes.oneOf([
        'successful',
        'failed', 'error',
        'running',
        'new', 'waiting', 'pending', 'canceled'
    ])
};

export default JobStatus;
