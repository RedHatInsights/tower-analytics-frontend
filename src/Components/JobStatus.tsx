import React, { FunctionComponent } from 'react';
import { Label } from '@patternfly/react-core';
import {
  CheckCircleIcon,
  ExclamationCircleIcon,
  ClockIcon,
  SyncAltIcon,
  ExclamationTriangleIcon,
} from '@patternfly/react-icons';

import { capitalize } from '../Utilities/helpers';

interface Props {
  status:
    | 'successful'
    | 'failed'
    | 'running'
    | 'pending'
    | 'error'
    | 'canceled'
    | 'new'
    | 'waiting';
}

const JobStatus: FunctionComponent<Props> = ({ status }) => {
  const getColor = () => {
    switch (status) {
      case 'successful':
        return 'green';
      case 'failed':
      case 'error':
        return 'red';
      case 'running':
      case 'pending':
        return 'blue';
      case 'canceled':
        return 'orange';
      // case new, waiting
      default:
        return 'grey';
    }
  };

  const getIcon = () => {
    switch (status) {
      case 'successful':
        return <CheckCircleIcon />;
      case 'failed':
      case 'error':
        return <ExclamationCircleIcon />;
      case 'running':
        return <SyncAltIcon />;
      case 'canceled':
        return <ExclamationTriangleIcon />;
      // case new, waiting, pending
      default:
        return <ClockIcon />;
    }
  };

  return (
    <Label variant="outline" color={getColor()} icon={getIcon()}>
      {capitalize(status)}
    </Label>
  );
};

export default JobStatus;
