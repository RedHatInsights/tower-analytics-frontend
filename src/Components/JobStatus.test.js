import React from 'react';
import { shallow } from 'enzyme';
import JobStatus from './JobStatus';

describe('Components/JobStatus', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = null;
  });

  it('should render successful', () => {
    wrapper = shallow(<JobStatus status="successful" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render failed', () => {
    wrapper = shallow(<JobStatus status="failed" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render error', () => {
    wrapper = shallow(<JobStatus status="error" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render running', () => {
    wrapper = shallow(<JobStatus status="running" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render new', () => {
    wrapper = shallow(<JobStatus status="new" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render waiting', () => {
    wrapper = shallow(<JobStatus status="waiting" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render pending', () => {
    wrapper = shallow(<JobStatus status="pending" />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render canceled', () => {
    wrapper = shallow(<JobStatus status="canceled" />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
