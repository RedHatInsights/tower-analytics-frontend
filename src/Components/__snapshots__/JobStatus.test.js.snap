// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/JobStatus should render canceled 1`] = `"<span class=\\"pf-c-label pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z\\" transform=\\"\\"></path></svg></span>Canceled</span></span>"`;

exports[`Components/JobStatus should render error 1`] = `"<span class=\\"pf-c-label pf-m-red pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z\\" transform=\\"\\"></path></svg></span>Error</span></span>"`;

exports[`Components/JobStatus should render failed 1`] = `"<span class=\\"pf-c-label pf-m-red pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z\\" transform=\\"\\"></path></svg></span>Failed</span></span>"`;

exports[`Components/JobStatus should render new 1`] = `"<span class=\\"pf-c-label pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z\\" transform=\\"\\"></path></svg></span>New</span></span>"`;

exports[`Components/JobStatus should render pending 1`] = `"<span class=\\"pf-c-label pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z\\" transform=\\"\\"></path></svg></span>Pending</span></span>"`;

exports[`Components/JobStatus should render running 1`] = `"<span class=\\"pf-c-label pf-m-blue pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z\\" transform=\\"\\"></path></svg></span>Running</span></span>"`;

exports[`Components/JobStatus should render successful 1`] = `"<span class=\\"pf-c-label pf-m-green pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z\\" transform=\\"\\"></path></svg></span>Successful</span></span>"`;

exports[`Components/JobStatus should render waiting 1`] = `"<span class=\\"pf-c-label pf-m-outline\\"><span class=\\"pf-c-label__content\\"><span class=\\"pf-c-label__icon\\"><svg style=\\"vertical-align:-0.125em\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z\\" transform=\\"\\"></path></svg></span>Waiting</span></span>"`;
